### PR TITLE
DM-13306 Multiple partitions test

### PR DESCRIPTION
**Pull request by mtpatter**
Add multiple partitions results section.

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -460,7 +441,85 @@ to be accessible from Kafka, and a slow consumer can have additional processing
 can slow the process, but the transit and consumption time should still be considered.  Consumer reading of the
 stream can easily be parallelized for faster processing.  The parallelization is controlled by the number of
 topic partitions, so increasing the number of partitions can decrease consumer read time such that consumers
-do not lag behind bursts if the consumers read in consumer groups in parallel.  For the critical time from
+do not lag behind bursts if the consumers read in consumer groups in parallel, which we explore below.
+For the critical time from
```
Comment (morriscb) [inline index.rst:445]: Looks like you have a extra carriage return here.

File: index.rst [hunk 2]
```diff
@@ -460,7 +441,85 @@ to be accessible from Kafka, and a slow consumer can have additional processing
...
+The alert stream we are using is pushed to a single "topic" within Kafka, into which alert messages are
+queued.  Kafka topics can then be divided into a number of "partitions," each of which can be placed on
+separate machines to allow for multiple consumers to read from a topic in parallel.
+We constructed a testbed for multiple partitions using an AWS Docker Swarm cloud of 3 Swarm managers
+on m4.16xlarge instances (64 vCPUs, 256 GiB memory, and 10,000 Mbps EBS bandwidth)
+and 10 worker nodes on m4.4xlarge instances (16 vCPUs, 64 GiB memory, and 2000 Mbps EBS bandwidth) all with
+1 TiB of ephemeral storage, as before.  This simulation used a Kafka cluster of 3 brokers configured to have
+an alert stream topic with 10 partitions, allowing consumers to parallelize into 10 consumer feeds.
+The Kafka containers were constrained to manager nodes and separated from the other alert stream
+producing and consuming containers. We used 100 containers for producers, each producing 100 alerts in
```
Comment (morriscb) [inline index.rst:462]: Looks like you are double spacing after a period but "containers. We" isn't. I'll check for others but I may miss them.

File: index.rst [hunk 3]
```diff
@@ -460,7 +441,85 @@ to be accessible from Kafka, and a slow consumer can have additional processing
...
+bursts every 39 seconds.  On the consumer side, we used 10 consumer groups each printing every 100th alert
+and 1 consumer group dropping alerts and printing only end of partition notices
+(i.e., a message that there are no more alerts to read) for monitoring.  Each consumer group consisted
+of 10 individual consumers acting in parallel to consume the full stream.
+
+Results
+^^^^^^^
+
+Using a 3-broker Kafka cluster, multiple partitions, and consumer groups decreases the total time alert
```
Comment (morriscb) [inline index.rst:471]: "total time alert distribution time" reads a bit awkwardly. I think I know what you mean maybe "Total time to distribute alerts"?

File: index.rst [hunk 4]
```diff
@@ -460,7 +441,85 @@ to be accessible from Kafka, and a slow consumer can have additional processing
...
+distribution time from initial Avro serialization of alerts to consumer receipt.  No significant difference
+from previous experiments of the compute resource utilization is observed.  The impact observed in
+previous simulations of an increased number of consumers slowing the producing time is lessened.
```
Comment (morriscb) [inline index.rst:474]: "producing time" -> "time to produce alerts"?

File: index.rst [hunk 5]
```diff
@@ -460,7 +441,85 @@ to be accessible from Kafka, and a slow consumer can have additional processing
...
+
+Conclusions
+-----------
+
+TO BE EXTENDED AND ORGANIZED IN TICKET DM-13307.
+
+For most experiments and metrics measured, assuming that the size of alerts has been estimated accurately,
+there is a significant impact on Kafka compute utilization and
+timing when postage stamp cutouts are included in the alert packets.
+The inclusion of postage stamp cutouts is currently a science requirement, however,
```
Comment (morriscb) [inline index.rst:514]: This sentence may do with chopping up into 2.

File: index.rst [hunk 6]
```diff
@@ -460,7 +441,85 @@ to be accessible from Kafka, and a slow consumer can have additional processing
...
+results here suggest that additional consideration be
+given to alternatives for the distribution of stamps, if it is possible
+that the current requirements may be adjusted.  One alternative to including postage
+stamps in the stream, recommended by other users of Kafka for large messages, is to forego including the
+stamps in the stream and instead include a url to the location of the stamps from which the stamp can be
+accessed separately.  This would particularly alleviate the increased network traffic for each additional
+consumer reading a stream.  Another potentially possible alternative could be to separate the postage stamps
+into their own Kafka topic.  If not all primary consumers are required or expected to read postage stamp
```
Comment (morriscb) [inline index.rst:522]: Split the text after the last comma into its own sentence.

**Review by morriscb — APPROVED** [review]
Looks good. Reads fairly clear except for a few cases mentioned below. Feel free to finish the pull after addressing the comments to your satisfaction.

The main points are possibly breaking up a few sentences and making the "time" you refer to in Results a bit more clear (e.g. total time alert distribution time). There are also a few periods that have inconsistent spacing after them.

File: index.rst (hunk 1 above)
Comment (mtpatter) [reply index.rst:445]: fixed.

File: index.rst (hunk 2 above)
Comment (mtpatter) [reply index.rst:462]: fixed.

File: index.rst (hunk 3 above)
Comment (mtpatter) [reply index.rst:471]: Yeah, you're right.  I'm struggling with the wording of that everywhere.  Fixed.  Thanks for the suggestion.

File: index.rst (hunk 4 above)
Comment (mtpatter) [reply index.rst:474]: Changed.

File: index.rst (hunk 5 above)
Comment (mtpatter) [reply index.rst:514]: Yeah, you're right.  Changed.

File: index.rst (hunk 6 above)
Comment (mtpatter) [reply index.rst:522]: Ok, fixed.

**Pull request merged**